Buffs and fixes for biogen!

Possumongus opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
Removed all non produce making animals or critters.
Buffed the fabrication time of chickens
Chickens can now be fabricated by the biogenerator.

Added recipes to make different kinds of meats in the bio generator

fixed duplicated entries in the bio generator

## Why / Balance
The biogenerator promises to be the new way to run service, however with generation times so long it made it hard to run monkeys, cows, etc and other kind of animals you would like to have in a kitchen. Now you can make a diverse set of meats to suit different menus, while also allowing you to keep making animals that produce milk or eggs

Duplicated entries in the biogen had no reason to exist, looked choppy and buggy

## Technical details
Removed duplicated entries and all non essential cubes from the bio generator
added non-salvageable meats to the bio generator
chickens can now be made in the generator at a reduced time

## How to test
Spawn a biogen
Check recipes, search "raw" if you wanna see all the meats
Search chicken to check for chicken cubes
try and fabricate a chicken to see times


## Media


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [Not revelant] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
Changed recipes on the bio generator, removed cubes that weren't relevant to its function.


**Changelog**
:cl:
- added: The biogenerator can now print more animal cubes and meat types.
- tweak: Biogenerator recipes generally print quicker but are more expensive.

dvir001 commented:
Ah yes print monkey in 3 sec to crash the server speed run,
This will be a problem, also you can upgrade the machine to gain faster crafting using machine upgrades.

Possumongus commented:
> Ah yes print monkey in 3 sec to crash the server speed run, This will be a problem, also you can upgrade the machine to gain faster crafting using machine upgrades.

no, the monkey cube and kobold cude print speed is not 3, those 2 recipes are duplicate entries that are not used, the issue is present in the vanilla files. the kobold and monkey cube you see in the biogenerator use the friendlycube list, I kept them since changing base ss14 files is iffy. but I could see to getting those entries removed as they are useless

dvir001 commented on Resources/Prototypes/Recipes/Lathes/Packs/biogen.yml:
```diff
@@ -11,8 +11,10 @@
+  #Frontier changes start
+  #- BioGenMonkeyCube
+  #- BioGenKoboldCube
+  #End of Frontier changes
```
Reformat this

# - BioGenMonkeyCube # Frontier: Duplicated
# - BioGenKoboldCube # Frontier: Duplicated

ErhardSteinhauer reviewed (changes requested):
TBC the reason why print times for cubes are so long is to avoid animal spam.

I will now use my powers to enforce this:
1. Remove the animal cubes save for chicken and, heavens help you if you do so, cow; print times for those should remain high (300 for a cow is fine, 180 for chicken seems a fair compromise).
2. Replace removed cubes with their meat equivalent, cost approximate to the current meat recipe. This allows for a pathway to access penguin/crab/etc options for specific meals that might be otherwise challenging to source.
3. No bear/kangaroo/carp meats (source the meat from wreck hoppers).

arimah commented:
> Remove the animal cubes save for chicken and, heavens help you if you do so, cow

We should keep the goat cube too. Only source of goats milk.

But keep in mind, with us gradually removing vending machines from food ships, freezers with butchering equipment will be far less useful if you can just wait a little while for meat to be printed. This represents a pretty significant change. :D

ErhardSteinhauer commented:
> butchering equipment will be far less useful if you can just wait a little while for meat to be printed.

True. Perhaps wild space animals should be changed so butchering them would require the spike to keep the equipment piece relevant. But it's outside the scope of this PR.

Possumongus commented:
> I will now use my powers to enforce this:
> 
>     1. Remove the animal cubes save for chicken, goat and, heavens help you if you do so, cow; print times for those should remain high (300 for a cow is fine, 180 for chicken seems a fair compromise).
> 
>     2. Replace removed cubes with their meat equivalent, cost approximate to the current meat recipe. This allows for a pathway to access penguin/crab/etc options for specific meals that might be otherwise challenging to source.
> 
>     3. No bear/kangaroo/carp meats (source the meat from wreck hoppers).
> 
> 
> Edit: goat cube should also stay.

So this changes the PR quite a lot so I did the following

Kept cow and goat as 300 seconds
chicken is 180
kept the critter cubes but at 180 themselves
removed all other cubes, replaced them with their meat equivalents

ErhardSteinhauer commented:
> So this changes the PR quite a lot so I did the following
> 
> Kept cow and goat as 300 seconds chicken is 180 kept the critter cubes but at 180 themselves removed all other cubes, replaced them with their meat equivalents

Changes look good to me (except for the critters like mothroach cubes production times, which is too low for my taste, but that can be adjusted later, if it becomes a problem), and works in the game. Gotta fix that submodule, it shouldn't be included in PR.

Possumongus commented:
> > So this changes the PR quite a lot so I did the following
> > Kept cow and goat as 300 seconds chicken is 180 kept the critter cubes but at 180 themselves removed all other cubes, replaced them with their meat equivalents
> 
> Changes look good to me (except for the critters like mothroach cubes production times, which is too low for my taste, but that can be adjusted later, if it becomes a problem), and works in the game. Gotta fix that submodule, it shouldn't be included in PR.

fixed the issues

whatston3 reviewed (changes requested):
Want to give this a review pass.

Possumongus commented:
> Want to give this a review pass.

sure, what do I need to do!

Possumongus commented:
> Given the bug reports and feedback issues that have cropped up in addition the the feel of how the system is working at present my sense is that it would be good to remove animal cube production entirely from the bio generator and consider an alternative source to cubes.

oh there are bugs, I did not see them, is there a place I can look for them?
I think I could try removing the cubes, is all in agreement?

Possumongus commented:
Working on conflicts, will get em done by tonight

Possumongus commented:
merging conflicts should be resolved

gitjubx commented:
Might aswell make the biogen able to suck in crops with magnet instead of just biomass so you dont need to single click each fruits into it helps aswell reduce lag , another thing since you want to buff it would be reducing the cost of flatpacked soils 1.500 is ...crazy so much infact noone ever makes even a single one ( 300-500 biomass MAX) its a cheap soil, science can make 10x more and better trays compared to cheap soils

gitjubx commented:
i think you have no idea or never tested how long and how many plants you would need to get a single soil so i recomment you doing it

Possumongus commented:
working on the magnet right now. Gonna change the soil stuff back to a higher speed

Possumongus commented:
okay I figured out a way to make the magnet work but its a TON of work and tagging and making sure I have all the plants produce and everything correctly setup.
I will keep this pr as is, but I'll change the dirt time to double the current rate.
Again I do feel it is useless and you can get much better results with hydroponic trays but yeah.

whatston3 commented:
Put up a suggestion branch.
- Restores monkey/kobold cubes (you removed them entirely)
- Makes the base cube take 180 seconds (actually 540 base, reasonable)
- Removed the cost for cheaper items (7 apples is fine, personal balance whinge to follow)
- Adds a category for animal cubes
- Adds a bunch of other cubes for zookeeper-ish roles

https://github.com/Possumongus/frontier-station-14/compare/biogenfixes...whatston3:2025-04-02-biogenfixes-suggestions?expand=1

Good to go from my side with these changes.

whatston3 commented:
The largest concern that I have with the biogenerator as-is balance-wise is that it's only one of three sources of biomass that we have - this, the biomass reclaimer, and the material reclaimer (with erhard's recycling bases).

Of those three, the material reclaimer gives back one biomass per recycled object (seed/produce), the biomass reclaimer gives back biomass proportional to the mass of the input, but only works on animals and requires a long time to run, and the biogenerator _absolutely spews out biomass_ from individual produce scaled on nutriment _instantly_.  This would be less of an issue if it never exited the machine, but apparently this is bad, so we need to balance around it.

In terms of a balance pass, both this and the material reclaimer should have relative parity considering they process the same stuff - a singular apple should not give you 10 biomass in one spot and 1 in another.  The costs should be relatively consistent as well. This PR as-is addresses _none of that_, only focusing on time.

Lastly, the times printed out on the lathe are a lie, it takes three times as long to print as what it claims - this could be fine but should be baked into the lathe itself so the reported times on the UI are accurate - I don't think that they'll show any differences for using space lube either.  This shouldn't be changed here, but should be a separate thing.

Ideally the lathes print out the amount of time something will take, both in the tooltip and in the queue, and you can cancel a given job in the queue when pressed.

Possumongus commented:
Oh forgot, on the note regarding time, does anyone have any idea why that is? I'm looking at the files and no obvious answer is springing up to mind.

Possumongus commented:
my complete apologies, I misread the feedback completely, changes implemented whatston, should be good?

whatston3 commented:
> I'm gonna do a pass on the biomass reclaimer to fix up the amount of bioproduct it gives, make it equal at least to current calculations on the biogen. There's also the option of just forgoing making biomass in the generator and using a reclaimer as the intended way of doing it.

I think there's value in the biogenerator being usable as a way to recycle produce, I just don't think it should have a drastically different return rate for what it processes, and I think that biomass should have a _reasonably consistent_ rate of return depending on what's being processed.  Makes sense for another go, this one's in the bag.

Thanks for the feature.

P.S. How many apples are you worth, exactly? :)

whatston3 reviewed (approved):
I think this is fine, but I'll leave this up for a while in case somebody else disagrees with which animals are generally printable.

Possumongus commented:
updated the changelog a bit to reflect new changes

whatston3 commented:
> updated the changelog a bit to reflect new changes

Reduced it to 2 points from 5 - look fine?

Possumongus commented:
looks  perfect, still do think we should note we removed the duplicated entry on the biogen!
so we end with 3 points
that sound good?

Possumongus commented:
well, then its good uhh... what now
